x/supply/types/token_holder.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// TokenHolder defines the interface used for modules that are allowed to hold
+// tokens. This is designed to prevent held tokens to be kept in regular Accounts,
+// as those are only ment for user accounts
+//
+// The bank module keeps track of each of the module's holdings and uses it to
+// calculate the total supply
+type TokenHolder interface {
```

Comment:
`TokenHolder` seems a bit...too loose. Does this pertain to only modules or also accounts? Perhaps it should be called `ModuleAccount` or `ModuleTokenHolder`?